engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityManager.java

```diff
@@ -575,18 +504,13 @@ public void notifyComponentRemovalAndEntityDestruction(long entityId, EntityRef
     @Override
     //Todo: be able to save components for entities in any pool
```

Comment:
Is this TODO item fixed with the new implementation? it looks like it will select the correct entity pool for the given id.